jina/serve/runtimes/gateway/http/models.py

```diff
@@ -266,6 +267,20 @@ class JinaRequestModel(BaseModel):
         description=DESCRIPTION_PARAMETERS,
     )
 
+    def dict(self, *args, **kwargs):
+        """
+        Convert to dict.
+
+        :param *args: args passed to BaseModel.dic()
+        :param *kwargs: kwargs passed to BaseModel.dic()
+        :return: self as Python dict
+        """
+        d = super().dict(*args, **kwargs)
+        if self.data is not None:
+            docs_dicts = DocumentArray.from_pydantic_model(self.data).to_dict()
+            d['data'] = docs_dicts
+        return d
```

Comment:
It feels a bit iffy here to create a DocumentArray just to then convert it to dict, but I can't really think of a nicer way.
Context: Here `self.data` is a `PydanticDocumentArray`, and in that the blob is always encoded *twice*. To handle properly, `from_pydantic_model` is used instead of a built in conversion do dict.